### ISSUE TITLE
When focusing terminal input input box, screen reader speaks "Terminal 1, occurred x times"

### ISSUE THREAD

**Issue by MarcoZehe** [microsoft/vscode#95511]
- VSCode Version: 1.45 Insider from April 17, 2020
- OS Version: Windows 10 19H2

Steps to Reproduce:

1. With NVDA running, open VS Code and open a terminal window. NVDA will speak "Terminal 1".
2. Press Shift+Tab to land on the "focus grab" div, then tab again to focus the input box. Result: NVDA speaks "Terminal 1, occurred again".
3. Repeating step 2 will increase the announcement of "occurred again" to "occurred x times", the counter always going up.

Does this issue occur when all extensions are disabled?: Yes

CC @isidorn

**Comment by isidorn** [comment]
I can reproduce this also with VoiceOver on my mac and with Orca.

So you move focus away from the terminal input box and then move focus back into the terminal and the Screen Reader announces each time that the terminal has focus. I do not funlly understand what is unexpected here? @MarcoZehe what would be the ideal behavior, that nothing gets read, or that occurred again part does not get read? I believe this is controlled by the screen readers and not sure what we can do here

fyi @leonardder @joanmarie

**Comment by MarcoZehe** [comment]
Well, I get two announcements. One for "Terminal 1", or whichever terminal console this pertains to, and this gets an announcement of "occurred again". I have never heard this "occurred again", or "occurred x times" anywhere else than in VS Code, so I assumed this came from VS Code, not from the screen reader.

The other announcement I get is the actual focusing of the input field, which is a slightly different announcement, and which does not have that "occurred again" announcement. The latter is probably the regular announcing of focused item, with name, role etc., the former some live region that announces that terminal 1 or 2 or whichever is active.

**Comment by isidorn** [comment]
@MarcoZehe you are correct, it does come from vscode [here](https://github.com/microsoft/vscode/blob/aecef14e24b4c3643bb6025c75819be4a608be1f/src/vs/base/browser/ui/aria/aria.ts#L73). So it looks like the terminal uses our `alert` call which it should not. Investigating..

**Comment by isidorn** [comment]
Oh so the terminal uses an explicit `alert` call [here](https://github.com/microsoft/vscode/blob/aecef14e24b4c3643bb6025c75819be4a608be1f/src/vs/workbench/contrib/terminal/browser/terminalTab.ts#L275)
@Tyriar why do we use an explicit `alert`? If I remove that code the screen reader nicely reads "Termian input" which you set [here](https://github.com/microsoft/vscode/blob/aecef14e24b4c3643bb6025c75819be4a608be1f/src/vs/workbench/contrib/terminal/browser/terminalInstance.ts#L336)

So I suggest to simply remove the alert call.
Idealy the aria-label which you set should also have the terminal name in it so we do not lose anything by removing the `alert` call.

**Comment by Tyriar** [comment]
I think the only reason for the alert is because we needed to read info about the terminal, so aria-label on the textarea is the way to go here?

**Comment by Tyriar** [comment]
I also notice this is printing the tab id, not the terminal ID. So all terminals in the first tab say they're terminal 1.

**Comment by isidorn** [comment]
@Tyriar yeah absolutelty go with the aria-label. That's a general use of thumb - try not to use alert ever.

**Comment by Tyriar** [comment]
Yeah this works much better, fix coming in a sec.

**Comment by isidorn** [comment]
Awesome, thanks a lot!

**Comment by jhomme** [comment]
Hi,
Could it be the status line firing?

**Comment by isidorn** [comment]
@jhomme no it is not the status bar, the cause was how this was imlemented, more details can be found in our comments above. THanks

**Comment by jvesouza** [comment]
It seems to have been fixed.

**Comment by isidorn** [comment]
@jvesouza thanks. Adding verified label.

**Comment by jhomme** [comment]
Hi,

The same thing happens for me, but it says a different message in another place. If I search for extensions on the market place and type slowly, the phrase “occurs n times,” where n is a number keeps speaking. I am thinking that it appears in the status line. I will check next time I open VSCode.

**Comment by Tyriar** [comment]
> The same thing happens for me, but it says a different message in another place. If I search for extensions on the market place and type slowly

@isidorn maybe another bug?

**Comment by isidorn** [comment]
I just checked the extensions search and there we use an alert once the marketplace results are fetched. I think this is fine and works decent in genereal
Though after there are results the aria label of the extensions search input should be updated
@jhomme can you please file a new issue for the extensions search and ping me on the issue? Thanks!
